ledger-service/http-json/src/main/scala/com/digitalasset/http/dbbackend/ContractDao.scala

```diff
@@ -100,6 +100,74 @@ object ContractDao {
     } yield domainContracts
   }
 
+  private[http] def selectContractsMultiTemplate[Pos](
+      parties: OneAnd[Set, domain.Party],
+      predicates: Seq[(domain.TemplateId.RequiredPkg, doobie.Fragment)],
+      trackMatchIndices: MatchedQueryMarker[Pos],
+  )(implicit
+      log: LogHandler
+  ): ConnectionIO[Vector[(domain.ActiveContract[JsValue], Pos)]] = {
+    import doobie.postgres.implicits._, cats.syntax.traverse._, cats.instances.vector._
+    predicates.zipWithIndex.toVector
+      .traverse { case ((tid, pred), ix) =>
+        surrogateTemplateId(tid) map (stid => (ix, stid, tid, pred))
```

Comment:
I always get scared of tuples with size > 3. Maybe worth considering introducing a case class for this but doesn’t have to be in this PR.